SalesforceCommerceCloud/pwa-kit
# @W-16797436@ [App Extensibility ⚙️] Allow overrides to be disabled

# Description

In this PR I ensure that any overrides that exist in a project DO NOT get used in the module resolution and building of the bundled code if they are DISABLED. This is mirror behaviour of how we handle the _Application Extension_ initialization. 

There is 1 caveat however, and that is that you'll most likely need to restart your dev server to see the effects of disabling extensions specifically with _overrides_ as the module resolution logic doesn't re-run. We'll make sure we document that in a known behaviours in a quip doc.

# Types of Changes




# Changes

- Filter out disabled _application extensions_ when generating lists of possible modules paths in `buildCandidatePaths`
- Added example extension that will be removed before merging this PR.

# How to Test-Drive This PR

- Check out this code.
- Run the `typescript minimal` project and notice that there is a red border around the application.
- Now edit the package.json to "enable" the "@salesforce/extension-sample-with-overrides" extension by setting enabled to true.
- Restart the dev server.
- Notice that the border is now green because an override in that extension is now running.

## General
